packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -804,14 +802,8 @@ export function scheduleUpdateOnFiber(
 
     if (root === workInProgressRoot) {
       // Received an update to a tree that's in the middle of rendering. Mark
-      // that there was an interleaved update work on this root. Unless the
-      // `deferRenderPhaseUpdateToNextBatch` flag is off and this is a render
-      // phase update. In that case, we don't treat render phase updates as if
-      // they were interleaved, for backwards compat reasons.
-      if (
-        deferRenderPhaseUpdateToNextBatch ||
-        (executionContext & RenderContext) === NoContext
-      ) {
+      // that there was an interleaved update work on this root.
+      if ((executionContext & RenderContext) === NoContext) {
```

Comment:
(question): As much as comments were inserted in relation to this condition, what do you think about inserting a variable to store this condition? But with a nomenclature that fits the context.

Something like: conditionAutoExplanatory = (executionContext & RenderContext) === NoContext

More infos: [juntossomosmais/frontend-guideline#63-descriptive-validations-if](https://github.com/juntossomosmais/frontend-guideline#63-descriptive-validations-if)

PS: I'm new to open-source, so I don't know if I'm saying something wrong

Reply:
Yeah I think that's a good practice sometimes. In this particular case, though, I think the comment would be necessary to provide additional context even if there were a descriptive variable name. So if the comment is going to be there regardless I'd rather keep the check inline.

Reply:
Got it! Thanks for clarifying!